component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java

```diff
@@ -116,4 +127,49 @@ public static String getResidentIdpProperty(String tenantDomain, String property
         }
         return propertyValue;
     }
+
+    /**
+     * Check whether the user store is based on identity data store.
+     *
+     * @return true if the user store is based on identity data store.
+     * @throws AuthenticationFailedException if an error occurs while initializing the UserStoreBasedIdentityDataStore.
+     */
+    public static boolean isUserStoreBasedIdentityDataStore() throws AuthenticationFailedException {
+
+        try {
+            String storeClassName = IdentityUtil.readEventListenerProperty(USER_OPERATION_EVENT_LISTENER_TYPE,
+                            IdentityStoreEventListener.class.getName()).getProperties()
+                    .get(DATA_STORE_PROPERTY_NAME).toString();
+            Class clazz = Class.forName(storeClassName.trim());
+            Object identityDataStore = clazz.newInstance();
+            return identityDataStore instanceof UserStoreBasedIdentityDataStore;
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new AuthenticationFailedException("Error while initializing the UserStoreBasedIdentityDataStore", e);
+        }
+    }
+
+    /**
+     * Check whether the user store is based on Active Directory.
+     *
+     * @param userStoreManager The user store manager.
+     * @return true if the user store is based on Active Directory.
+     */
+    public static boolean isActiveDirectoryUserStore(UserStoreManager userStoreManager) {
+
+        return userStoreManager instanceof UniqueIDJDBCUserStoreManager
+                && userStoreManager.getSecondaryUserStoreManager() instanceof UniqueIDActiveDirectoryUserStoreManager;
+    }
+
+    /**
+     * Convert Windows file time to Unix time.
+     *
+     * @param windowsFileTime Windows file time.
+     * @return Unix time.
+     */
+    public static String convertWindowsFileTimeToUnixTime(String windowsFileTime) {
+
+        long fileTime = Long.parseLong(windowsFileTime);
+        long millisSinceEpoch = (fileTime / HUNDREDS_OF_NANOSECONDS) - WINDOWS_EPOCH_DIFF;
```

Comment:
Let's add an explanatory comment on the conversion between Windows File time and Unix time.